### PR TITLE
Randomize zookeeper port for test

### PR DESCRIPTION
**Pull request by xiangfu0**
So all the tests should be using randomized ports.

### CODE REVIEW

File: pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java [hunk 1]
```diff
@@ -181,9 +181,15 @@ public ControllerRequestClient getControllerRequestClient() {
     return _controllerRequestClient;
   }
 
+  public void startDefaultTestZk() {
```
Comment (walterddr) [inline pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java:184]: who is using this new method?

File: pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java [hunk 2]
```diff
@@ -181,9 +181,15 @@ public ControllerRequestClient getControllerRequestClient() {
...
+    if (_zookeeperInstance == null) {
+      _zookeeperInstance = ZkStarter.startLocalZkServer(ZkStarter.DEFAULT_ZK_TEST_PORT);
+    }
+  }
+
   public void startZk() {
     if (_zookeeperInstance == null) {
-      _zookeeperInstance = ZkStarter.startLocalZkServer();
+      _zookeeperInstance = ZkStarter.startLocalZkServer(NetUtils.findOpenPort(20000 + RandomUtils.nextInt(10000)));
```
Comment (walterddr) [inline pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java:192]: any specific reason to find a port between 20000 and 30000?

Comment (xiangfu0) [reply pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java:192]: no, just random picked.

File: pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java (hunk 1 above)
Comment (xiangfu0) [reply pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java:184]: not yet, in case there is a need later

**Review by walterddr — APPROVED** [review]

**Review by Jackie-Jiang — COMMENTED** [review]
Do we need this? We are already using random port in `ZkStarter.startLocalZkServer()` starting from the default port.
Regarding the recently merged random port for other components, do we start from the existing default port and find the first available port after the default? That will help us finding the port when using the test in debug mode

**Comment by xiangfu0** [comment]
Just realized the port is already randomized. Closed it for now.

**Pull request closed without merging**